packages/framework/src/Model/Product/ProductRepository.php

```diff
@@ -165,10 +157,12 @@ public function addTranslation(QueryBuilder $queryBuilder, $locale)
      * @param \Doctrine\ORM\QueryBuilder $queryBuilder
      * @param int $domainId
      */
-    public function addDomain(QueryBuilder $queryBuilder, $domainId)
+    private function addDomain(QueryBuilder $queryBuilder, $domainId)
     {
-        $queryBuilder->join(ProductDomain::class, 'pd', Join::WITH, 'pd.product = p AND pd.domainId = :domainId');
-        $queryBuilder->setParameter('domainId', $domainId);
+        $queryBuilder->addSelect('pd')
+            ->join('p.domains', 'pd', Join::WITH, 'pd.domainId = :domain');
+
+        $queryBuilder->setParameter('domain', $domainId);
```

Comment:
I do not understand changes in this method

Reply:
ProductRepository does some visibility calculations in which they control ProdudctDomain properties, thats why it is needed to be added. It is same concept as it is used for Translations.